cmd/cdi-importer/importer.go

```diff
@@ -61,9 +63,15 @@ func main() {
 	}
 
 	dest := common.ImporterWritePath
-	if contentType == string(cdiv1.DataVolumeArchive) {
+	if volumeMode == string(v1.PersistentVolumeBlock) &&
+		contentType == string(cdiv1.DataVolumeKubeVirt) && source == controller.SourceHTTP {
+		dest = common.ImporterWriteBlockPath
+	}
+
+	if contentType == string(cdiv1.DataVolumeArchive) || source == controller.SourceRegistry {
```

Comment:
This should be:
```go
if contentType == string(cdiv1.DataVolumeArchive {
  dest = common.ImporterVolumePath
}
```
Otherwise any registry targets will be a directory instead of a file.

Reply:
done